charts/rancher-csp-adapter/0.1.0+up0.1.0-rc2/templates/serviceAccount.yaml

```diff
@@ -0,0 +1,17 @@
+{{- if eq (include "csp-adapter.csp" . ) "aws" -}}
+  {{- if eq (include "csp-adapter.awsValuesSet" .) "true" -}}
+  {{- else -}}
+  {{- fail "If the aws provider is enabled, account number and role name must be provided" -}}
+  {{- end -}}
+{{- else -}}
+{{- fail "One cloud provider must be enabled" -}}
+{{- end -}}
```

Comment:
Very cool, I like this system. Curious how it will display in dashboard.

Reply:
One note here - a failure at this point will fail to install the chart (i.e. will provide an error message on the cli where the user is attempting to install). There won't be an error in the dashboard (unless they try to install through the UI or something). The dashboard-visible errors require the csp-adapter to be successfully installed to be present.